sphinx/ext/autodoc.py

```diff
@@ -508,6 +513,36 @@ def parse_name(self):
                         (self.objpath and '.' + '.'.join(self.objpath) or '')
         return True
 
+    def monkey_patch_import_function(self):
+        """Replace the builtin ``import`` function.
+
+        When a module cannot be imported, and its name begins by something it
+        ``autodoc_mock_imports``, return a dummy one which makes the
+        auto-documented modules "importable".
+        """
+        def _import(name, globals=None, locals=None, fromlist=(), level=None):
+            try:
+                if PY2:
+                    if globals is None:
+                        globals = {}
+                    if locals is None:
+                        locals = {}
+                    if level is None:
+                        level = -1
+                elif level is None:
+                    level = 0
+                return REALIMPORT(name, globals, locals, fromlist, level)
+            except ImportError:
+                if any(name.startswith(m)
```

Comment:
If you want I could make sure that all mocked imports are cleaned up when done processing the autodoc.